cmd/sriovdp/manager.go

```diff
@@ -290,3 +279,58 @@ type linkWatcher struct {
 func (lw *linkWatcher) Subscribe() {
 
 }
+
+func (rm *resourceManager) netDevHandler(dev *ghw.PCIDevice) {
+	// exclude device in-use in host
+	if isDefaultRoute, _ := hasDefaultRoute(dev.Address); !isDefaultRoute {
+
+		aPF := utils.IsSriovPF(dev.Address)
+		aVF := utils.IsSriovVF(dev.Address)
+
+		if aPF || !aVF {
+			// add to linkWatchList
+			rm.addToLinkWatchList(dev.Address)
+		}
+
+		if aPF && utils.SriovConfigured(dev.Address) {
+			// do not add this device in net device list
+			return
+		}
+
+		if newDevice, err := resources.NewPciNetDevice(dev, rm.rFactory); err == nil {
+			rm.netDeviceList = append(rm.netDeviceList, newDevice)
+		} else {
+			glog.Errorf("discoverDevices() error adding new device: %q", err)
+		}
+
+	}
+}
+
+func (rm *resourceManager) accDevHandler(dev *ghw.PCIDevice) {
+	//Check if device is a physical function (PF)
+	aPF := utils.IsSriovPF(dev.Address)
+	devID, err := strconv.ParseInt(dev.Product.ID, 16, 64)
+	if err != nil {
+		glog.Warningf("discoverDevices(): unable to parse device ID for device %+v %q", dev, err)
+		return
+	}
+	//Check if device ID = 0b30, it means it is a RSU (Remote System Update) device and not accelerator
```

Comment:
It seems messy to need to be worrying about vendor-specific PCI device IDs in the middle of this code. Is there nothing else we can look at to determine the class of a device, or to make this more generic so that adding other accelerators would be simpler?